streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -257,7 +258,7 @@ private boolean setState(final State newState) {
                 return false;
             } else if (!state.isValidTransition(newState)) {
                 log.error("Unexpected state transition from {} to {}", oldState, newState);
```

Comment:
Do we really want to log and throw? I beleive this will get logged again where it is caught, so it seems a bit odd to log it here. 
Also just realised this hasn't changed in this PR, but still...

Reply:
IllegalStateException will not be caught within the library, but I agree that when it was thrown all the way up and stop the JVM, this log entry will be printed again.

Thinking about this a bit, I'd probably just add back the prefix and then remove the logging.